src/main/resources/liquibase/crop_changelog/18_3_0.xml

```diff
@@ -0,0 +1,400 @@
+			INSERT INTO cvterm (cvterm_id, cv_id, name, definition, dbxref_id, is_obsolete, is_relationshiptype)
+			VALUES (4031, 1020, 'Assigned Attribute', 'Standard method for assigned attribute', null, 0, 0);
+        </sql>
+    </changeSet>
+
+	<changeSet author="clarysabel" id="v18.3.0-4">
+		<preConditions onFail="MARK_RAN">
+			<sqlCheck expectedResult="0">select count(*) from cvterm where cvterm_id = 4032;</sqlCheck>
+		</preConditions>
+		<comment>Add 'Assigned Passport Descriptor' method</comment>
+        <sql>
+			INSERT INTO cvterm (cvterm_id, cv_id, name, definition, dbxref_id, is_obsolete, is_relationshiptype)
+			VALUES (4032, 1020, 'Assigned Passport Descriptor', 'Standard method for assigned passport descriptor', null, 0, 0);
+        </sql>
+    </changeSet>
+
+	<changeSet author="clarysabel" id="v18.3.0-5">
+		<preConditions onFail="MARK_RAN">
+			<not><tableExists tableName="udflds_attributes_bkp"></tableExists></not>
+		</preConditions>
+		<comment>Create properties and insert values in udflds_attributes_bkp in preparation for the attributes migration</comment>
+		<sql>
+			CREATE TABLE udflds_attributes_bkp (
+			  fldno            int,
+			  ftable           varchar(24),
+			  ftype            varchar(12),
+			  fcode            varchar(50),
+			  fname            varchar(50),
+			  ffmt             varchar(255),
+			  fdesc            varchar(255),
+			  lfldno           int,
+			  fuid             int,
+			  fdate            int,
+			  scaleid          int,
+			  cvterm_name      varchar(200),
+			  property_id      int,
+			  method_id        int,
+			  variable_type_id int,
+			  cvterm_id        int
+			);
+
+			INSERT INTO udflds_attributes_bkp (fldno, ftable, ftype, fcode, fname, ffmt, fdesc, lfldno, fuid, fdate, scaleid, cvterm_name, method_id, variable_type_id)
+			SELECT fldno, ftable, ftype, fcode, fname, ffmt, fdesc, lfldno, fuid, fdate, scaleid, CONCAT(REPLACE(fcode, ' ', ''), '_A', SUBSTRING(ftype, 1, 1), '_text'), IF(ftype = 'PASSPORT', 4032, 4031), IF(ftype = 'PASSPORT', 1813, 1814)
+			FROM udflds WHERE ftable = 'ATRIBUTS' AND ftype IN ('ATTRIBUTE', 'PASSPORT', 'METHOD', 'CHARACTERIZATION', 'USE');
+
+			CREATE TEMPORARY TABLE new_properties
+			SELECT DISTINCT bkp.fname name
+			FROM udflds_attributes_bkp bkp
+			WHERE NOT EXISTS(SELECT name
+							 FROM cvterm
+							 WHERE cv_id = 1010 AND name = bkp.fname AND is_obsolete = 0);
+
+			INSERT INTO cvterm (cv_id, name, definition, dbxref_id, is_obsolete, is_relationshiptype)
+			SELECT 1010, name, name, null, 0, 0 FROM new_properties;
+
+			INSERT INTO cvterm_relationship(type_id, subject_id, object_id)
+			SELECT 1225, cvt.cvterm_id, 1380 from new_properties tmp INNER JOIN cvterm cvt ON (cvt.cv_id = 1010 AND
+			cvt.name = tmp.name AND cvt.is_obsolete = 0);
+
+			DROP TEMPORARY TABLE new_properties;
+
+			UPDATE udflds_attributes_bkp bkp INNER JOIN cvterm cvterm ON bkp.fname = cvterm.name
+			SET bkp.property_id = cvterm.cvterm_id WHERE cvterm.cv_id = 1010;
```

Comment:
if there are _obsolete_ properties with the same name, they'll be created:
```sql
...
WHERE NOT EXISTS(SELECT name
	FROM cvterm
	WHERE cv_id = 1010 AND name = bkp.fname AND is_obsolete = 0)
```
 which I guess is fine, but further below where  it queries again to update _bkp.property_id_, it doesn't check for _is_obsolete = 0_:

```sql
UPDATE udflds_attributes_bkp bkp INNER JOIN cvterm cvterm ON bkp.fname = cvterm.name
SET bkp.property_id = cvterm.cvterm_id WHERE cvterm.cv_id = 1010;
```
Could that happen and be a problem? could the update fails because the join returns more than one row? @clarysabel

Reply:
I will add the is_obsolete condition. I did not find any case with is_obsolete being != 0 but given that the unique key is name, cv_id and is_obsolete I added the condition to all the queries. I missed this one.